### PR TITLE
Collaborators tests

### PR DESCRIPTION
**Pull request by alopezsanchez**
- `lib/collaborators` tests.
- `reorganice fixtures with a README subfolder`. I created some README files in order to test all possible throw cases.

I also have a question: should [`isCollaborator function`](https://github.com/joyeecheung/node-core-utils/blob/master/lib/collaborators.js#L104) return `undefined`/`collaborator` or it should return `true`/`false` instead?

### CODE REVIEW

File: test/unit/collaborators.test.js [hunk 1]
```diff
@@ -1,16 +1,120 @@
 'use strict';
 
+const assert = require('assert');
+
 const {
+  isCollaborator,
   getCollaborators
 } = require('../../lib/collaborators');
-const { readme, collaborators } = require('../fixtures/data');
+const { readme, readmeNoTsc, readmeNoTscE, readmeNoCollaborators, readmeNoCollaboratorE, readmeUnordered, collaborators } = require('../fixtures/data');
```
Comment (joyeecheung) [inline test/unit/collaborators.test.js]: Can you wrap this line by breaking after the commas?

File: test/unit/collaborators.test.js [hunk 2]
```diff
@@ -1,16 +1,120 @@
...
+  });
+
+  afterEach(() => {
+    logger.clear();
+  });
+
+  describe('Collaborator', () => {
+    describe('isActor', () => {
+      it('should return false if no actor is specified', () => {
+        assert.strictEqual(collaborator.isActor(), false);
```
Comment (joyeecheung) [inline test/unit/collaborators.test.js]: This should be `collaborator.isActor(null)`, github returns the actor as `null` if it's a ghost.

File: test/unit/collaborators.test.js [hunk 3]
```diff
@@ -1,16 +1,120 @@
...
+    logger = new TestLogger();
+  });
+
+  afterEach(() => {
+    logger.clear();
+  });
+
+  describe('Collaborator', () => {
+    describe('isActor', () => {
+      it('should return false if no actor is specified', () => {
```
Comment (joyeecheung) [inline test/unit/collaborators.test.js]: `...if actor is a ghost`

File: test/unit/collaborators.test.js [hunk 4]
```diff
@@ -1,16 +1,120 @@
...
+  });
+
+  describe('getCollaborators', () => {
+    it('should return all collaborators', async function() {
+      const parsed = await getCollaborators(readme, logger, 'nodejs', 'node');
+      assert.deepStrictEqual(parsed, collaborators);
+    });
+
+    it('should throw error if there is no TSC section in the README', async() => {
+      await assertThrowsAsync(async() => getCollaborators(readmeNoTsc, logger, 'nodejs', 'node'), /Error: Couldn't find ### TSC \(Technical Steering Committee\) in the README/);
```
Comment (joyeecheung) [inline test/unit/collaborators.test.js]: Can you wrap the arguments passed to `assertThrowsAsync`? (and the lines below)

File: test/unit/collaborators.test.js [hunk 5]
```diff
@@ -1,16 +1,120 @@
...
+    });
+
+    it('should throw error if there are no collaborators', async() => {
+      await assertThrowsAsync(async() => getCollaborators(readmeUnordered, logger, 'nodejs', 'node'), /Error: Could not find any collaborators/);
+    });
+  });
+
+  describe('isCollaborator', () => {
+    it('should return undefined if the user has no login (ghost)', () => {
+      assert.strictEqual(isCollaborator(collaborators, { name: 'Ghost' }), undefined);
```
Comment (joyeecheung) [inline test/unit/collaborators.test.js]: If it's a ghost then the whole thing would be `null`, the login check is just a safeguard, I don't think Github would ever return an actor without `login`..

**Comment by alopezsanchez** [comment]
Updated with requested changes. I also applied the max line length = 80 (https://github.com/joyeecheung/node-core-utils/pull/81)

File: test/unit/collaborators.test.js [hunk 6]
```diff
@@ -1,16 +1,173 @@
...
+    it('should return all collaborators', async function() {
+      const parsed = await getCollaborators(readme, logger, 'nodejs', 'node');
+      assert.deepStrictEqual(parsed, collaborators);
+    });
+
+    it(
+      'should throw error if there is no TSC section in the README',
+      async() => {
+        await assertThrowsAsync(
+          async() => getCollaborators(readmeNoTsc, logger, 'nodejs', 'node'),
```
Comment (cPhost) [inline test/unit/collaborators.test.js]: A minor nit: can you add space between `async` and `()`

File: test/unit/collaborators.test.js [hunk 7]
```diff
@@ -1,16 +1,173 @@
...
+          Error);
+      });
+
+    it(
+      'should throw error if there is no TSC Emeriti section in the README',
+      async() => {
+        await assertThrowsAsync(
+          async() => getCollaborators(readmeNoTscE, logger, 'nodejs', 'node'),
```
Comment (cPhost) [inline test/unit/collaborators.test.js]: ^ same as above

File: test/unit/collaborators.test.js [hunk 8]
```diff
@@ -1,16 +1,173 @@
...
+            readmeUnordered, logger, 'nodejs', 'node'),
+          /Error/);
+        assert.strictEqual(logger.logs.warn[0][0],
+          'Order of different types of contacts needs update. ' +
+          'Analysis could get wrong.');
+      });
+
+    it('should throw error if there are no collaborators', async() => {
+      await assertThrowsAsync(
+        async() => getCollaborators(readmeUnordered, logger, 'nodejs', 'node'),
```
Comment (cPhost) [inline test/unit/collaborators.test.js]: Same as above^ for all.

File: test/unit/collaborators.test.js (hunk 6 above)
Comment (apapirovski) [reply test/unit/collaborators.test.js]: That wouldn't pass the linter, the current version is correct.

**Pull request merged**